internal/pkg/agent/application/fleet_server_bootstrap_test.go

```diff
@@ -141,6 +143,51 @@ func TestInjectFleetConfigComponentModifier(t *testing.T) {
 
 }
 
+func TestFleetConfigComponentModifier_MultipleHosts(t *testing.T) {
+	cfg := &configuration.FleetServerConfig{
+		Output: configuration.FleetServerOutputConfig{
+			Elasticsearch: configuration.Elasticsearch{
+				Protocol: "https",
+				Hosts:    []string{"test-server:9200"},
+			},
+		},
+	}
+	log, _ := logger.NewTesting("test")
+	modifier := FleetServerComponentModifier(cfg, log)
+
+	fleetServerOutputSource, err := structpb.NewStruct(map[string]interface{}{
+		"type":    "elasticsearch",
+		"api_key": "example",
+		"hosts":   []interface{}{"https://localhost:9200", "https://test-server:9200"},
+	})
+	require.NoError(t, err)
+	fleetServerComponent := component.Component{
+		InputSpec: &component.InputRuntimeSpec{
+			InputType: "fleet-server",
+		},
+		Units: []component.Unit{
+			{
+				Type: client.UnitTypeOutput,
+				Config: &proto.UnitExpectedConfig{
+					Type:   "elasticsearch",
+					Source: fleetServerOutputSource,
+				},
+			},
+		},
+	}
+	comps := []component.Component{fleetServerComponent}
+	resComps, err := modifier(comps, nil)
+	require.NoError(t, err)
+	t.Logf("RESCOMPS %v", resComps)
+	require.Len(t, resComps, 1)
+	require.Len(t, resComps[0].Units, 1)
+
+	resCfg := resComps[0].Units[0].Config.GetSource().AsMap()
+	assert.NotContains(t, resCfg, "api_key")
+	assert.Contains(t, resCfg, "hosts")
+	assert.Len(t, resCfg["hosts"], 2)
```

Comment:
[Blocker]
Why didn't you assert the new `hosts` have the right hosts there? Currently if there is any bug which causes the wrong hots to be used, as log as it's the right amount of hosts the test will not catch it.